extensions-core/multi-stage-query/src/main/java/org/apache/druid/msq/querykit/results/ExportResultsFrameProcessorFactory.java

```diff
@@ -127,11 +139,46 @@ public ProcessorsAndChannels<Object, Long> makeProcessors(
     );
 
     return new ProcessorsAndChannels<>(
-        ProcessorManagers.of(processors),
+        ProcessorManagers.of(processors)
+                         .withAccumulation(new ArrayList<String>(), (acc, file) -> {
+                           ((ArrayList<String>) acc).add((String) file);
+                           return acc;
+                         }),
         OutputChannels.none()
     );
   }
 
+  @Nullable
+  @Override
+  public TypeReference<Object> getResultTypeReference()
+  {
+    return new TypeReference<Object>() {};
+  }
+
+  @Override
+  public Object mergeAccumulatedResult(Object accumulated, Object otherAccumulated)
+  {
+    // Maintain upgrade compatibility, if a worker does not return a list, ignore it.
```

Comment:
Can we end up in a state where some files are present and some are not? What is to say that there are files absent? I think if the manifest file is for cosmetic purposes only, we can get away with partial results, but it should be explicitly called out in the documentation. 

Since it's in a widely used format, the expectation seems to be that the users would be able to ingest the results in other systems, like:
Druid -> Export destination -> External system

This will lead to incorrect results, without the users knowing. I think we should fail the job here, and if not, then this seems like the usage of the manifest file is limited by its correctness and should be called out explicitly. We can also have a context parameter (which I am not a big fan of), that can switch between error-throwing and ignoring behaviour, for those who want correctness.

Reply:
Changed to throw an exception